### ISSUE TITLE
List.Lazy insertAt, modifyAt, updateAt, deleteAt, alterAt members behave differently to List and Array

### ISSUE THREAD

**Issue by garyb** [purescript/purescript-lists#28]
They always return a `List a` even when the index is out of range, whereas the other modules return a `Maybe (List a)` or `Array (List a)` for when the operation has no effect.

**Comment by paf31** [comment]
The reason is that we don't have a lazy `Maybe`. We would need at least to return `Lazy (Maybe (List a))` unless we wanted to force the head element.

**Comment by garyb** [comment]
Yeah, I wondered if this one might be problematic. It can always go ignored if we decide its not worth it :smile:

**Comment by JordanMartinez** [comment]
This needs further discussion.
